drivers/interrupt_controller/intc_renesas_ra_icu.c

```diff
@@ -26,99 +30,112 @@ enum {
 	IELSRn_OFFSET = 0x300,
 };
 
-int ra_icu_query_exists_irq(uint32_t event)
+static inline unsigned int event_to_irq(unsigned int event)
 {
-	for (uint32_t i = 0; i < CONFIG_NUM_IRQS; i++) {
-		uint32_t els = sys_read32(IELSRn_REG(i)) & UINT8_MAX;
-
-		if (event == els) {
-			return i;
-		}
+	if (event >= sizeof(function_irq_table)) {
```

Comment:
It would be better to use ARRAY_SIZE here instead of sizeof.

Reply:
Fixed it. Thank you!